fuse_optimizers/src/batch_optimizer.cpp

```diff
@@ -206,12 +212,15 @@ void BatchOptimizer::transactionCallback(
       start_time_ = transaction_time;
     }
     // Purge old transactions from the pending queue
-    ros::Time purge_time(0, 0);
+    rclcpp::Time purge_time(0, 0, transaction_clock_type);  // NOTE(CH3): Uninitialized
     if (started_)
     {
       purge_time = start_time_;
     }
-    else if (ros::Time(0, 0) + params_.transaction_timeout < last_pending_time)  // prevent a bad subtraction
+    // NOTE(CH3): In this case we're okay with uninitialized time since it's just used
+    //            for comparison
+    // prevent a bad subtraction
+    else if (rclcpp::Time(0, 0, transaction_clock_type) + params_.transaction_timeout < last_pending_time)
```

Comment:
I'm not sure why `rclcpp::Time(0,0)` is being added. Does `else if (params_.transaction_timeout < last_pending_time)` work?

Reply:
The timeout is a Duration, so the arithmetic is to get a Time object offset by the duration for comparison with last_pending_time, which is also a Time object.

Reply:
https://github.com/locusrobotics/fuse/pull/283/commits/c16b02f15441790cdeea5bcd023940801ca89a01